cli/lsp/language_server.rs

```diff
@@ -2795,21 +2806,26 @@ impl Inner {
       vec![(referrer.clone(), deno_graph::ModuleKind::Esm)]
     };
 
-    if self.maybe_cache_server.is_none() {
-      self.maybe_cache_server = Some(
-        cache::CacheServer::new(
-          self.maybe_cache_path.clone(),
-          self.maybe_import_map.clone(),
-          self.maybe_config_file.clone(),
-          None,
-          None,
-          None,
-        )
-        .await,
-      );
-    }
-    let cache_server = self.maybe_cache_server.as_ref().unwrap();
-    if let Err(err) = cache_server.cache(roots).await {
+    let mut cli_options = CliOptions::new(
+      Flags {
+        cache_path: self.maybe_cache_path.clone(),
+        ca_stores: None,
+        ca_file: None,
+        unsafely_ignore_certificate_errors: None,
+        ..Default::default()
+      },
+      self.maybe_config_file.clone(),
+    );
+    cli_options.set_import_map_specifier(self.maybe_import_map_uri.clone());
+
+    // todo(dsherret): why is creating a basic runtime necessary? It errors otherwise.
+    let handle = std::thread::spawn(|| {
+      run_basic(
+        async move { create_graph_for_caching(cli_options, roots).await },
+      )
+    });
```

Comment:
I'm not sure why this is necessary, but without it, it errors. Maybe this is why the `CacheServer` was initially made, but this code here is much simpler.

Reply:
When you create a new thread there's no running Tokio runtime there, so you can't use async function because there is no executor for futures you're trying to run. Try removing new thread creation and see if it works

Reply:
Actually, I shouldn't block the main thread (not sure what thread is running this, but probably the main thread). I should spawn a blocking task and await it.

Reply:
I think there might have been some deadlock happening if it was run asynchronously on the same thread, but I'm not sure. Have you tried moving it back to the same thread to see if everything works as expected?